File: Programmers/C++/Level1/Caesar_Cipher.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    
    for(int i = 0; i < s.length(); i++)
    {
        if(s[i] != ' ')
        {
            for(int j = 0; j < n; j++)
            {
                if(s[i] == 'z') s[i] = 96;
                if(s[i] == 'Z') s[i] = 64;
                s[i]++;
            }
            answer += s[i];
        }
        else answer += ' ';
    }
    
    return answer;
}

File: Programmers/Python/Level1/Caesar_cipher.py
upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" 
lower = "abcdefghijklmnopqrstuvwxyz"

def solution(s, n):
    answer = ''

    for i in s:
        if i == " ":
            answer += " "
        elif i.isupper():
            idx = upper.find(i) + n
            if idx > 25: idx -= 26
            answer += upper[idx]
        else:
            idx = lower.find(i) + n
            if idx > 25: idx -= 26
            answer += lower[idx]
        
    return answer